upgrade/upgrade-scripts/execute

```diff
@@ -69,92 +70,93 @@ EOF
+# of reasons:
 #
-#  - After finishing processing the dependencies of
-#    delphix-virtualization, Apt continues processing the other
-#    dependencies of delphix-entire-aws, including delphix-platform-aws.
+# 1. When moving from one "delphix-entire" version to another, it's
+#    possible that we'll want to "downgrade" some package that the
+#    "delphix-entire" package would depend on. Unfortunately, if we
+#    simply used "apt-get install delphix-entire", this would not allow
+#    us to install the "downgraded" version of the package.
 #
-#  - Apt sees that delphix-platform-aws conflicts with
-#    delphix-platform-kvm, so it un-marks delphix-platform-kvm for
-#    installation, allowing delphix-platform-aws to satisfy
-#    virtualization's dependency on delphix-platform instead. However,
-#    it does _not_ unmark all of the dependencies of
-#    delphix-platform-kvm, so some unnecessary kvm-related packages end
-#    up being installed, such as linux-image-4.15.0-1028-kvm.
+#    For example, consider the following situation:
 #
-# In short, we can end up with some unwanted packages installed because
-# Apt marked and then unmarked some package for installation while
-# trying to satisfy a dependency on a virtual package.
+#        - "delphix-entire" is currently installed at version 2.0
+#        - "zfs" is currently installed at version 2.0
+#        - "delphix-entire" version 3.0 depends on "zfs" version 1.0
 #
-# To avoid this we install the delphix-platform-<platform> package that
-# we want before installing delphix-entire-<platform>. If some
-# delphix-platform package is already installed, that dependency is
-# already satisfied and Apt doesn't need to choose an arbitrary one to
-# satisfy the dependency.
+#    If we tried to "apt-get install delphix-entire=3.0" it would fail
+#    because it could not install "zfs" version 1.0, since that'd be a
+#    downgrade of the "zfs" package. This is because "apt-get install"
+#    doesn't support downgrading dependent packages, only packages
+#    explicitly specified on the command line; e.g. we'd have to run the
+#    command "apt-get install delphix-entire=3.0 zfs=1.0" for this to
+#    work.
 #
-# In general, we don't want upgrade to need know too much about the set
-# of packages to be installed. However, it seems reasonable to make an
-# exception for delphix-platform because delphix-platform is a
-# fundemental piece of the architecture of the appliance. Among other
-# things, it is the mechanism that we use to orchestrate the installation
-# all of the platform-dependant bits of the appliance.
+# 2. Often times, developers want to build and install upgraded packages
+#    that "delphix-entire" would depend on. Since the dependencies of
+#    the "delphix-entire" package would have to depend on explicit
+#    versions of each package, this would prevent packages from being
+#    upgraded without either:
 #
-apt_get install -y "delphix-platform-$platform" ||
```

Comment:
I think we might have problems if we remove the explicit installation of delphix-platform. We have some dependencies where delphix-platform and its packages must be configured before installing other packages (e.g. delphix-platform creates the postgres user on postinst which should be done before installing postgres).

Also I don't believe that installing each package explicitly is any different from installing it through delphix-entire as a dependency, thus if we needed to install delphix-platform first before, we'd still need it now.

Reply:
> Also I don't believe that installing each package explicitly is any different from installing it through delphix-entire as a dependency, thus if we needed to install delphix-platform first before, we'd still need it now.

@jgallag88 do you have any thoughts on if we still need to explicitly install `delphix-platform` with these changes? is there any way to test and verify if it's still needed?

I'll plan on keeping it since I'm not really sure either way, but it'd be nice to know.

Reply:
It's not clear to me. Apt might or might not treat `apt-get install foo` the exactly same `apt-get install bar baz` with respect to how it resolves dependencies if `foo` depends on `bar` and `baz`. However, it doesn't make any guarantees either way, so it seems safer to keep the explicit installation of `delphix-platform`, like you have it now.

As for testing, I was just grepping through the `apt-get` for packages for other kernels (`*kvm*`, `*azure*`, etc) to see if they were selected at any point.